stdlib/public/Concurrency/AsyncThrowingStream.swift

```diff
@@ -12,61 +12,203 @@
+    /// A type that indicates how the stream terminated.
+    ///
+    /// The `onTermination` closure receives an instance of this type.
     public enum Termination {
       
-      /// The stream was finished via the `finish` method
+      /// The stream finished as a result of calling the continuation's
+      ///  `finish` method.
+      ///
+      ///  The associated `Failure` value provides the error that terminated
+      ///  the stream. If no error occurred, this value is `nil`.
       case finished(Failure?)
       
-      /// The stream was cancelled
+      /// The stream finished as a result of cancellation.
       case cancelled
     }
     
-    /// A result of yielding values.
+    /// A type that indicates the result of yielding a value to a client, by
+    /// way of the continuation.
+    ///
+    /// The various `yield` methods of `AsyncThrowingStream.Continuation` return
+    /// this type to indicate the success or failure of yielding an element to
+    /// the continuation.
     public enum YieldResult {
       
-      /// When a value is successfully enqueued, either buffered
-      /// or immediately consumed to resume a pending call to next
-      /// and a count of remaining slots available in the buffer at
-      /// the point in time of yielding. Note: transacting upon the
-      /// remaining count is only valid when then calls to yield are
-      /// mutually exclusive.
+      /// The stream successfully enqueued the element.
+      ///
+      /// This value reprsents the successful enqueueing of an element, whether
+      /// the stream buffers the element or delivers it immediately to a pending
+      /// call to `next()`. The associated value `remaining` is a hint that
+      /// indicates the number of remaining slots in the buffer at the time of
+      /// the `yield` call.
+      ///
+      /// - Note: From a thread safety viewpoint, `remaining` is a lower bound
+      /// on the number of remaining slots. This is because a subsequent call
+      /// that uses the `remaining` value could race on the consumption of
+      /// values from the stream.
       case enqueued(remaining: Int)
       
-      /// Yielding resulted in not buffering an element because the
-      /// buffer was full. The element is the dropped value.
+      /// The stream didn't enqueue the element due to a full buffer.
+      ///
+      /// The associated element for this case is the element that the stream
+      /// dropped.
       case dropped(Element)
       
-      /// Indication that the continuation was yielded when the
-      /// stream was already in a terminal state: either by cancel or
-      /// by finishing.
+      /// The stream didn't enqueue the element because the stream was in a
+      /// terminal state.
+      ///
+      /// This indicates the stream terminated prior to calling `yield`, either
+      /// because the stream finished normally or through cancellation, or
+      /// threw an error
```

Comment:
```suggestion
      /// threw an error.
```

Should that be "it threw an error" or "through an error"? Please fix accordingly.

Reply:
Errors are things you `throw`, so "threw" here is correct. Might be worth adding "it", to clarify the stream is what throws the error.